package.json

```diff
@@ -50,7 +50,7 @@
   },
   "husky": {
     "hooks": {
-      "pre-commit": "npm run format"
+      "pre-commit": "git diff --name-only HEAD | grep '.*\\.js' | xargs prettier --write"
```

Comment:
At work we do this weird thing where we don't allow partial commits for precommit hooks. I don't know if there's merit to doing it that way or not, but there are a few things I'm suspicious of here:

1. We should probably only run this on staged changes with the `--cached` flag so that we don't accidentally overwrite files that are known to be yet ready for commit.
2. Does this `git diff --name-only HEAD | grep '.*\\.js'` thing work for you? I think I understand the purpose of the double backslash (escape string, then regexp) but why is there a period at the beginning of this?

Reply:
`.*` means any character, repeated any number of times. I agree on the --cached being better. I'm not super worried thought because it just runs Prettier.